R/checkDates.R

```diff
@@ -0,0 +1,72 @@
+#' Check if time period of BAIT input data (rsds, sfc, huss) is congruent with
+#' near-surface temperature data (tas).
+#'
+#' @param baitInput list of raster data encompassing different climate variables
+#' @param tasData raster data on near-surface atmosphere temperature
+#'
+#' @return baitInput with congruent time periods w.r.t. tasData
+#'
+#' @importFrom terra rast subset
+#' @importFrom stringr str_sub
+
+checkDates <- function(baitInput, tasData) {
+  datesT <- names(tasData)
+
+  baitInput <- sapply(names(baitInput), function(var) { # nolint
```

Comment:
There actually is a good reason to avoid `sapply` as it can return all sorts of objects which can be dangerous. Have you tried replacing it with `lapply`? I guess you want a named list. I think you can return a named list of length one in `lapply` and get the same.
```suggestion
  baitInput <- lapply(names(baitInput), function(var) {
```